CDCgov/prime-reportstream
# 7402 - Added support for ingesting FHIR bundle through reports endpoint

This PR allows for FHIR bundles to be passed in through the reports endpoint. This for now is using the same FHIR that ReportStream would normally deal with from after the convert step on HL7, and doesn't mean we can broadly accept any FHIR input.

Test Steps:

1. Send a POST request to send a FHIR report
    - Using Postman
        - [Test 7402.postman_collection.json](https://github.com/CDCgov/prime-reportstream/files/10145306/Test.7402.postman_collection.txt)
    - Via curl command
        - [temp.txt](https://github.com/CDCgov/prime-reportstream/files/10145318/temp.txt)

3. Verify that the request was accepted
4. Look at the database to verify that the FHIR report has gone through the expected actions (look at reports and lineage)

## Changes
- Edited `organizations.yml` to add sender for `ignore-fhir` (plus many automatic formatting updates)
- Added a mimeType for `Sender.Format.FHIR` and split out the behavior in `ELRReceiver` depending on the format of the sender
- Added a mimeType for `Report.Format.FHIR` and a function that can return a `Report.Format` from a file extension. Also a function which can return a `Report.Format` from a `blobURL`
- Split out HL7-specific parts of the convert step into a separate function, and make a similar function for FHIR-specific parts of the convert step
- Updated the FHIR converter to read the format from the blobURL and call the appropriate function for the format-specific parts of the convert step
- Added unit tests

## Checklist

### Testing

## Linked Issues
- Resolves #7402
